spec/models/mixins/authentication_mixin_spec.rb

```diff
@@ -721,4 +728,76 @@ def self.name; "TestClass"; end
       end
     end
   end
+
+  describe "#authentication_status_severity_level" do
+    it "uses the least valid status" do
+      EvmSpecHelper.local_miq_server
+      authentications
+      expect(host.authentication_status_severity_level.status).to eq("Invalid")
+    end
+  end
+
+  describe "#authentication_status" do
+    before do
+      EvmSpecHelper.local_miq_server
+    end
+
+    context "with no authentications" do
+      before { host }
+
+      it "is nil with sql" do
+        expect(virtual_column_sql_value(Host, "authentication_status")).to be_nil
+      end
+
+      it "is 'None' with pure ruby (via relations)" do
+        expect(host.authentication_status).to eq("None")
+      end
+
+      it "is 'None' when accessed via ruby, but fetched via sql" do
```

Comment:
Worth nothing:  This spec and the one above have a important distinction to make here since it is slightly confusing when looking at the result that is provided by `virtual_column_sql_value`.  The default value of `"None"` is desired here as we want to maintain the previous functionality when calling `authentication_status`.